### ISSUE TITLE
Yak-40 by Felis. It's not possible to set parking brake after towing is completed [X-Plane 10].

### ISSUE THREAD

**Issue by McDonne11** [skiselkov/BetterPushbackC#9]
Hello!
First of all, I would like to thank you for fixing crash in XP10.
It's just the best plugin!
But the following problem arose. Which I described in the title of the topic.
If you do not have this airplane, you can download it from the link below:
http://felis-planes.com/yak40/
Once again, thank you for the excellent plugin and for the timely responses! Regards!

**Comment by McDonne11** [comment]
The same problem with An-24 by Felis. I can Set/Release parking brake after tow, but no reaction from tug.

**Comment by skiselkov** [comment]
Sorry, I tried to analyze this, but the Felis is doing something really strange to the parking brake as soon as my plugin starts controlling the nosewheel. No idea how to fix it and I don't think Felis is going to make adjustments, given how old the aircraft models are.

**Comment by McDonne11** [comment]
Thanx for answer!

**Comment by parshukov** [comment]
as a developer of the patch 1 for An24 after Felis made this aircraft free and open source I can say - I will try to fix it ;)

**Comment by skiselkov** [comment]
Thank you! It shouldn't be too difficult. Is it using a custom parking brake dataref like the Tu-154M? If so, what is it? I can pick it up instead of the standard one inside the plugin.

**Comment by parshukov** [comment]
I will check it and write here

**Comment by skiselkov** [comment]
You can see which datarefs BP looks at here: https://github.com/skiselkov/BetterPushbackC/blob/master/src/bp.c#L575-L580

**Comment by parshukov** [comment]
try this one for an24
defineProperty("brake1", globalPropertyf("sim/flightmodel/controls/l_brake_add")) -- gear brake ratio. 0 = min, 1 = max
defineProperty("brake2", globalPropertyf("sim/flightmodel/controls/r_brake_add")) -- gear brake ratio. 0 = min, 1 = max
defineProperty("emerg_brake", globalPropertyf("sim/custom/xap/An24_hydro/emerg_brake"))  -- pressure in braking system
defineProperty("block_brake", globalPropertyf("sim/flightmodel/controls/parkbrake"))  -- blocks brakes
defineProperty("gear_blocks", globalPropertyi("sim/custom/xap/An24_covers/gear_blocks"))  -- gear blocks

**Comment by parshukov** [comment]
sorry. do not look at my last message, try this one:
sim/flightmodel/controls/parking_brake
0.70 - on
0- off

**Comment by parshukov** [comment]
Saso, do you create any datarefs for a pushback to know, that he is operating? I looked at Felis code of Yak40 and there is ............. won't say it))) for me it is easier to add into the logic some if-then-else so to disable logic while the pushback is operating)

**Comment by skiselkov** [comment]
Yes, you can look at the int `bp/started` dataref. It's set to 1 when the tug is moving around the aircraft. However, I remember you sending something about `sim/flightmodel/controls/parking_brake` - is this still correct? I can add an extra line that checks for the Yak-40 and uses that dataref for the parking brake instead of the normal one.

**Comment by parshukov** [comment]
Ok, I will implement it in to the aircraft logic. sim/flightmodel/controls/parking_brake is not on 100% correct, cause there is custom logic for braking system... thanks! I will use int` bp/started` and check it!

**Comment by parshukov** [comment]
well... I made some improvements into the code, so who wants to fly on Yak40 can download this file and change the existing in the Custom Avionics folder.
[brake_logic.zip](https://github.com/skiselkov/BetterPushbackC/files/1197116/brake_logic.zip)

**Comment by skiselkov** [comment]
Thanks, I'll give this one a try.

**Comment by skiselkov** [comment]
Unfortunately it seems I can't test it, both the Yak-40 and An-24 crash X-Plane 10 on my machine (Linux).

**Comment by parshukov** [comment]
Our community already tested it and it is OK :)

**Comment by skiselkov** [comment]
Thank you. Magically, I tried it again and all of a sudden, it was ok... :) So I can confirm that now as well.
Just to advise: I'm going to make one change. Currently BP is picking the Goldhofer AST-3F tug to push the Yak-40. That is incorrect, the Yak-40 has single-wheel nose gear and the AST-3F cannot handle that. I've made changes so that the Yak-40 is handled by the smaller LEKTRO AP88, which is more appropriate to this class of aircraft.
